### PR TITLE
check for notebook iframe since jupyter lab starts faster

### PR DESCRIPTION
**Pull request by sanderegg**

### CODE REVIEW

**Review by pcrespov — APPROVED** [review]

File: tests/e2e/tutorials/jupyters.js [hunk 1]
```diff
@@ -72,7 +78,13 @@ async function runTutorial() {
 
     const iframeHandles2 = await tutorial.getIframe();
     // expected three iframes = loading + jupyterNB + jupyterLab
-    const jLabIframe = await iframeHandles2[2].contentFrame();
+    const iFrame20 = await iframeHandles2[0].contentFrame();
+    const iFrame21 = await iframeHandles2[1].contentFrame();
+
+    var jLabIframe = iFrame20;
```
Comment (odeimaiz) [inline tests/e2e/tutorials/jupyters.js:84]:
```suggestion
    var jLabIframe = iFrame20;
```
    let jLabIframe = iFrame20;

**Review by odeimaiz — APPROVED** [review]

File: tests/e2e/tutorials/jupyters.js [hunk 2]
```diff
@@ -32,7 +32,13 @@ async function runTutorial() {
 
     const iframeHandles = await tutorial.getIframe();
     // expected two iframes = loading + jupyterNB
-    const nbIframe = await iframeHandles[1].contentFrame();
+    const iFrame0 = await iframeHandles[0].contentFrame();
+    const iFrame1 = await iframeHandles[1].contentFrame();
+
+    var nbIframe = iFrame0;
```
Comment (odeimaiz) [inline tests/e2e/tutorials/jupyters.js:38]:
```suggestion
    let nbIframe = iFrame0;
```

**Review by odeimaiz — APPROVED** [review]

File: tests/e2e/tutorials/jupyters.js [hunk 3]
```diff
@@ -32,7 +32,13 @@ async function runTutorial() {
...
+    if (iFrame1._url.endsWith("tree?")) {
+      nbIframe = iFrame1;
+    }
```
Comment (odeimaiz) [inline tests/e2e/tutorials/jupyters.js:41]:
```suggestion
    const nbIframe = [iFrame0, iFrame1].find(iframe => iframe._url.endsWith("tree?");
```

File: tests/e2e/tutorials/jupyters.js [hunk 4]
```diff
@@ -72,7 +78,13 @@ async function runTutorial() {
...
     const iframeHandles2 = await tutorial.getIframe();
     // expected three iframes = loading + jupyterNB + jupyterLab
-    const jLabIframe = await iframeHandles2[2].contentFrame();
+    const iFrame20 = await iframeHandles2[0].contentFrame();
+    const iFrame21 = await iframeHandles2[1].contentFrame();
+
+    var jLabIframe = iFrame20;
+    if (iFrame21._url.endsWith("lab?")) {
+      jLabIframe = iFrame21;
+    }
```
Comment (odeimaiz) [inline tests/e2e/tutorials/jupyters.js:87]:
```suggestion
    const jLabIframe = [iFrame20, iFrame21].find(iframe => iframe._url.endsWith("lab?");
```

**Review by odeimaiz — APPROVED** [review]

File: tests/e2e/tutorials/jupyters.js (hunk 4 above)
Comment (sanderegg) [reply tests/e2e/tutorials/jupyters.js:84]: very pretty!

**Pull request merged**